### ISSUE TITLE
Generator in FunctionExpression is not preserved.

### ISSUE THREAD

**Issue by mbostock** [Rich-Harris/butternut#43]
Input:

```js
foo = function*() {
  yield 42;
};
```

Output:

```js
foo=function(){yield 42}
```

This throws a SyntaxError: Unexpected strict mode reserved word because the `yield` is now in a normal (non-generator) function. The expected output is:

```js
foo=function*(){yield 42}
```

**Comment by Rich-Harris** [comment]
Thanks — looks like `async` functions are similarly broken, working on a fix

**Comment by Rich-Harris** [comment]
Fixed in 0.3.6
